src/accelerate/accelerator.py

```diff
@@ -1655,19 +1657,16 @@ def _prepare_ipex(self, *args):
             elif isinstance(obj, (torch.optim.Optimizer)):
                 optimizer = obj
         if optimizer is not None and model is not None:
-            if is_ipex_available():
-                if is_xpu_available() and self.device.type == "xpu":
-                    model = model.to(self.device)
-                    model, optimizer = torch.xpu.optimize(model, optimizer=optimizer, inplace=True, level="O1")
-                    model.forward = torch.xpu.amp.autocast()(model.forward)
-                else:
-                    model, optimizer = ipex.optimize(model, optimizer=optimizer, inplace=True, level="O1")
-                    model.forward = torch.cpu.amp.autocast()(model.forward)
+            dtype = self.state.ipex_plugin.dtype
+            if dtype == "no":
+                dtype = torch.float32
```

Comment:
The `dtype` should be the one of the Accelerator state. We shouldn't have two different dtypes as the user will not necessarily keep them in sync and this will result in subtle bugs.